python/lsst/ts/tcpip/mock_json_reading_one_client_server.py

```diff
@@ -0,0 +1,62 @@
+# This file is part of ts_tcpip.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import typing
+from collections import deque
+
+from .constants import LOCAL_HOST
+from .json_reading_one_client_server import JsonReadingOneClientServer
+
+__all__ = ["MockJsonReadingOneClientServer"]
+
+
+class MockJsonReadingOneClientServer(JsonReadingOneClientServer):
+    """An implementation of MessageReadingOneClientServer for unit test
+    purposes.
+    """
+
+    def __init__(self) -> None:
+        log = logging.getLogger()
+        super().__init__(
+            host=LOCAL_HOST,
+            port=0,
+            log=log,
+            name="MockJsonReadingOneClientServer",
+        )
+
+        self._json_dict_queue: deque = deque()
+        self.json_dict_event = asyncio.Event()
+
+    async def handle_json_dict(self, json_dict: dict[str, typing.Any]) -> None:
+        self._json_dict_queue.append(json_dict)
+        self.json_dict_event.set()
+
+    async def get_first_json_dict(self) -> dict[str, typing.Any]:
```

Comment:
In the same vein as earlier comments I suggest removing "json" from the method names and attribute names.

The important thing is that the queue accumulates data. How it was transmitted is an internal detail.